mesh-watcher/files/mesh-watcher.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+. /lib/functions.sh
+. /lib/functions/network.sh
+
+CHECK_TIME=$1
+CHECK_BABELD=$2
+
+babeld_get_mesh_interface() {
+	config_get $1 ifname
+}
+
+babeld_get_mesh_neighbors() {
+    config_load babeld
+    mesh_interfaces=$(config_foreach babeld_get_mesh_interface interface ifname)
+
+    for mesh in $mesh_interfaces; do
+        network_get_physdev phy $mesh
+        linklocal=$(ip -6 a list dev $phy | grep "scope link" | awk '{print $2}' | sed 's/\/64//')
+        ips=$(ping ff02::1%$phy -c2 | grep from | awk '{print $4}' | sed 's/.$//')
+        for ip in $ips; do
+            if [ $ip != $linklocal ] && [ $(owipcalc $ip linklocal) -eq 1 ]; then
+                echo $ip
```

Comment:
You can break the loops once you found a single one and save some CPU time by ending early. Maybe also rename the function to "has_neighbors".